packages/subgraph/src/packages/admin/adminMembers.ts

```diff
@@ -0,0 +1,63 @@
+import {dataSource, store} from '@graphprotocol/graph-ts';
+import {
+  Administrator,
+  AdministratorAdminPlugin
+} from '../../../generated/schema';
+import {Granted, Revoked} from '../../../generated/templates/DaoTemplate/DAO';
+
+export function handleGranted(event: Granted): void {
+  if (
+    isCorrectEvent(
+      event.params.permissionId.toHexString(),
+      event.params.where.toHexString()
+    )
+  ) {
+    let pluginAddress = event.params.where.toHexString();
+    let administratorAddress = event.params.who.toHexString();
+    let administrator = Administrator.load(administratorAddress);
+    if (!administrator) {
+      administrator = new Administrator(administratorAddress);
+      administrator.address = administratorAddress;
+      administrator.save();
+    }
+
+    let administratorMappingId = `${pluginAddress}_${administratorAddress}`;
+    let administratorPluginMapping = AdministratorAdminPlugin.load(
+      administratorMappingId
+    );
+    if (!administratorPluginMapping) {
+      administratorPluginMapping = new AdministratorAdminPlugin(
+        administratorMappingId
+      );
+      administratorPluginMapping.administrator = administratorAddress;
+      administratorPluginMapping.plugin = pluginAddress;
+      administratorPluginMapping.save();
+    }
+  }
+}
+
+export function handleRevoked(event: Revoked): void {
+  if (
+    isCorrectEvent(
+      event.params.permissionId.toHexString(),
+      event.params.where.toHexString()
+    )
+  ) {
+    let mappingId = `${event.params.where.toHexString()}_${event.params.who.toHexString()}`;
```

Comment:
why not removing `Administrator` entity as well ?

Reply:
Because the proposals are linked to the Administrator. If we remove this entity we don't have the Administrator entities for the proposals anymore